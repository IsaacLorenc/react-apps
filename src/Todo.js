import React from "react";

const Todo = ({ id, task, removeTodo, editTodo }) => {
  const handleRemove = () => removeTodo(id);
  const handleEdit = () => {
    const updatedTask = prompt("Enter new task:", task);
    if (updatedTask) {
      editTodo(id, updatedTask);
    }
  };

  return (
    <li>
      <div>{task}</div>
      <button onClick={handleEdit}>Edit</button>
      <button onClick={handleRemove}>X</button>
    </li>
  );
};

export default Todo;