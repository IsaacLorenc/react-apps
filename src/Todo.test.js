import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

test('renders without crashing', () => {
  render(<Todo id={1} task="Test Todo" removeTodo={() => {}} editTodo={() => {}} />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<Todo id={1} task="Test Todo" removeTodo={() => {}} editTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test('can remove todo', () => {
  const removeTodoMock = jest.fn();
  const { getByText } = render(<Todo id={1} task="Test Todo" removeTodo={removeTodoMock} editTodo={() => {}} />);
  const removeButton = getByText('X');

  fireEvent.click(removeButton);

  expect(removeTodoMock).toHaveBeenCalledWith(1);
});