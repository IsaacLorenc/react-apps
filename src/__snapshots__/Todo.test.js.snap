// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Test Todo
    </div>
    <button>
      Edit
    </button>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
